From a7cf113469ba32951a0cfa44a35992153ae876c8 Mon Sep 17 00:00:00 2001
From: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Date: Tue, 4 Jul 2017 07:57:19 +0200
Subject: [PATCH 33/33] UBUNTU-only: apparmor: for kvm.powerpc (LP: #1680384)

The (so far) Ubuntu only kvm wrappers call a lot more on ppc.
Since this is already considered as the qemu binary it must be opened up
in apparmor to work.
So allow these extra tools executed by kvm.powerpc

Forwarded: no (part of continuous upstreaming effort)
Signed-off-by: Christian Ehrhardt <christian.ehrhardt@canonical.com>
---
 examples/apparmor/libvirt-qemu | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/examples/apparmor/libvirt-qemu b/examples/apparmor/libvirt-qemu
index b8e3106..58ef8aa 100644
--- a/examples/apparmor/libvirt-qemu
+++ b/examples/apparmor/libvirt-qemu
@@ -181,6 +181,13 @@
   /sys/bus/ r,
   /sys/class/ r,
 
+  # kvm.powerpc executes/accesses this
+  /bin/uname rmix,
+  /usr/sbin/ppc64_cpu rmix,
+  /bin/grep rmix,
+  /sys/devices/system/cpu/subcores_per_core r,
+  /sys/devices/system/cpu/cpu*/online r,
+
   # Allow guests to receive signals from and be traced by
   # libvirtd (LP: #1298611). Requires apparmor >= 2.9.
   signal (receive) peer=/usr/sbin/libvirtd,
-- 
2.7.4

